src/go/k8s/api/redpanda/v1alpha1/redpanda_clusterspec_types.go

```diff
@@ -389,7 +389,7 @@ type Logging struct {
 	// Sets the verbosity level of logs.
 	LogLevel string `json:"logLevel"`
 	// Specifies whether to send usage statistics to Redpanda Data.
-	UsageStats UsageStats `json:"usageStats"`
```

Comment:
I don't think this will work as expected. `omitempty` doesn't control whether or not a field is required or not. It just controls whether or not it's present in the JSON marshalled version of the struct which doesn't work the way you'd think it would when it comes to struct fields.

We'd probably need to change this to be `*UsageStats` and also add a kubebuilder annotation. Though I also wonder if we should start importing the Values struct from the helm-chart into the operator at this point 🤔